src/rest.js

```diff
@@ -50,7 +61,11 @@ export class Rest {
         return response.json().catch(error => null);
       }
 
-      throw response;
+      if (parseError) {
+        return response.json().catch(() => response).then(error => Promise.reject(error));
```

Comment:
Or neither? It will already return a rejected promise. You can leave off the catch.